express-zod-api/src/common-helpers.ts

```diff
@@ -63,7 +63,11 @@ export const getInput = (
 };
 
 export const ensureError = (subject: unknown): Error =>
-  subject instanceof Error ? subject : new Error(String(subject));
+  subject instanceof Error
+    ? subject
+    : subject instanceof z.ZodError
+      ? new Error(getMessageFromError(subject), { cause: subject })
```

Comment:
this might no longer be needed since Colin reverted ZodError non-extending from Error

Reply:
it is, but the message is unreadable and has to be addressed this way